research-app-messages/src/selections.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Information about a selected source in the engine.
+ * The values for right ascension and declination are given in radians.
+ * `catalogName` gives the name of the associated HiPS catalog,
+ * while `name` gives a string name for the source.
+ */
+ export interface Source {
+    ra: number;
+    dec: number;
+    name: string;
+    catalogName: string;
```

Comment:
Here I think we need to declare a TypeScript indexer to indicate that there will be additional fields for the catalog columns, and mention that in the docstring.

It might be good to namespace the arbitrary catalog data more strongly (e.g., you can worry about potential issues about well-definedness if the catalog provides a column named `ra`, etc.) but we're mirroring the internal app data structures here, and I'd prefer not to spend too much time fiddling with those, and I think what we've got here will be fine.

Reply:
This potential issue of well-definedness of name is also present in the app itself. I agree that we don't want to spend too much time messing with the data structure in the app, but I'm thinking we could go with something along the lines of:

```
export interface Source {
    ra: number;
    dec: number;
    name: string;
    catalogName: string;
    info: {
        [field: string]: string | undefined;
    }
}
```

where the `info` object contains all of the catalog data.

Reply:
Yeah, that would be the most sensible way to encapsulate, I think. Maybe name the field `catalogData` or something. If you think it will be straightforward to modify the app's Source object this way too, I do think it would be good for future-proofiness.